### PR TITLE
session: add physical table placement_policy in mysql schema

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close part of #26580
 Design table in mysql.* for persisting placement policies

### What is changed and how it works?

What's Changed:
add physical table mysql.placement_policy to store the placement policy.
add this create table ddl both in the upgrade and bootstrap phase.

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test

### Release note 

```release-note
session: add physical table placement_policy in mysql schema.
```

### CODE REVIEW

**Comment by morgo** [comment]
This test is failing:
```sql
[2021-08-10T05:12:36.524Z] infoschema_reader_test.go:917:
[2021-08-10T05:12:36.524Z]     c.Assert(len(tk.MustQuery("select TABLE_NAME from information_schema.TABLE_STORAGE_STATS where TABLE_SCHEMA = 'mysql';").Rows()), Equals, 24)
[2021-08-10T05:12:36.524Z] ... obtained int = 25
[2021-08-10T05:12:36.524Z] ... expected int = 24
```

File: session/bootstrap.go [hunk 1]
```diff
@@ -339,6 +339,22 @@ const (
 		WITH_GRANT_OPTION enum('N','Y') NOT NULL DEFAULT 'N',
 		PRIMARY KEY (USER,HOST,PRIV)
 	  );`
+
+	// CreatePlacementPolicyTable store the placement policys.
+	CreatePlacementPolicyTable = `CREATE TABLE IF NOT EXISTS mysql.placement_policy (
+		NAME varchar(255) NOT NULL,
```
Comment (morgo) [inline session/bootstrap.go:345]: Tables/views are limited to 64. I think we should chose similar for consistency:
```suggestion
		NAME varchar(64) NOT NULL,
```

**Review by morgo — COMMENTED** [review]
Looking good, minor comment.

File: session/bootstrap.go (hunk 1 above)
Comment (AilinKid) [reply session/bootstrap.go:345]: Yes, sure

**Review by morgo — APPROVED** [review]

File: session/bootstrap.go [hunk 2]
```diff
@@ -339,6 +339,22 @@ const (
 		WITH_GRANT_OPTION enum('N','Y') NOT NULL DEFAULT 'N',
 		PRIMARY KEY (USER,HOST,PRIV)
 	  );`
+
+	// CreatePlacementPolicyTable store the placement policys.
+	CreatePlacementPolicyTable = `CREATE TABLE IF NOT EXISTS mysql.placement_policy (
+		NAME varchar(64) NOT NULL,
```
Comment (bb7133) [inline session/bootstrap.go:345]: It would be nice if you can add a comment for each of the columns.

**Review by xhebox — APPROVED** [review]

File: session/bootstrap.go [hunk 3]
```diff
@@ -339,6 +339,22 @@ const (
 		WITH_GRANT_OPTION enum('N','Y') NOT NULL DEFAULT 'N',
 		PRIMARY KEY (USER,HOST,PRIV)
 	  );`
+
+	// CreatePlacementPolicyTable store the placement policys.
+	CreatePlacementPolicyTable = `CREATE TABLE IF NOT EXISTS mysql.placement_policy (
```
Comment (djshow832) [inline session/bootstrap.go:344]: Why not treat it as a virtual table like `information_schema.tables`, and only store `placementPolicy` to the metadata, just like `tableInfo`?
The disadvantage of creating a real table is that it's really complex to update the data definition between versions.

**Pull request merged**